src/coreComponents/physicsSolvers/multiphysics/poromechanicsKernels/ThermalMultiphasePoromechanics_impl.hpp

```diff
@@ -0,0 +1,670 @@
+}
+
+template< typename SUBREGION_TYPE,
+          typename CONSTITUTIVE_TYPE,
+          typename FE_TYPE >
+GEOSX_HOST_DEVICE
+GEOSX_FORCE_INLINE
+void ThermalMultiphasePoromechanics< SUBREGION_TYPE, CONSTITUTIVE_TYPE, FE_TYPE >::
+computePoreVolumeConstraint( localIndex const k,
+                             real64 const & porosity,
+                             real64 const & dPorosity_dVolStrain,
+                             real64 const & dPorosity_dPressure,
+                             real64 const & dPorosity_dTemperature,
+                             StackVariables & stack ) const
+{
+  using Deriv = constitutive::multifluid::DerivativeOffset;
+
+  Base::computePoreVolumeConstraint( k,
+                                     porosity,
+                                     dPorosity_dVolStrain,
+                                     dPorosity_dPressure,
+                                     dPorosity_dTemperature,
+                                     stack );
+
+  arraySlice1d< real64 const, compflow::USD_PHASE - 1 > const phaseVolFrac = m_fluidPhaseVolFrac[k];
+  arraySlice2d< real64 const, compflow::USD_PHASE_DC - 1 > const dPhaseVolFrac = m_dFluidPhaseVolFrac[k];
+
+  stack.dPoreVolConstraint_dTemperature = 0.0;
+  for( integer ip = 0; ip < m_numPhases; ++ip )
+  {
+    stack.dPoreVolConstraint_dTemperature += -dPhaseVolFrac( ip, Deriv::dT ) * porosity
+                                             - phaseVolFrac( ip ) * dPorosity_dTemperature;
+  }
+}
+
+template< typename SUBREGION_TYPE,
+          typename CONSTITUTIVE_TYPE,
+          typename FE_TYPE >
+GEOSX_HOST_DEVICE
+GEOSX_FORCE_INLINE
+void ThermalMultiphasePoromechanics< SUBREGION_TYPE, CONSTITUTIVE_TYPE, FE_TYPE >::
+assembleMomentumBalanceTerms( real64 const ( &N )[numNodesPerElem],
+                              real64 const ( &dNdX )[numNodesPerElem][3],
+                              real64 const & detJxW,
+                              StackVariables & stack ) const
+{
+  using namespace PDEUtilities;
+
+  Base::assembleMomentumBalanceTerms( N, dNdX, detJxW, stack );
+
+  constexpr FunctionSpace displacementTrialSpace = FE_TYPE::template getFunctionSpace< numDofPerTrialSupportPoint >();
+  constexpr FunctionSpace displacementTestSpace = displacementTrialSpace;
+  constexpr FunctionSpace pressureTrialSpace = FunctionSpace::P0;
+
+  // compute local linear momentum balance residual derivatives with respect to temperature
+
+  BilinearFormUtilities::compute< displacementTestSpace,
+                                  pressureTrialSpace,
+                                  DifferentialOperator::SymmetricGradient,
+                                  DifferentialOperator::Identity >
+  (
+    stack.dLocalResidualMomentum_dTemperature,
+    dNdX,
+    stack.dTotalStress_dTemperature,
+    1.0,
+    -detJxW );
+
+  if( m_gravityAcceleration > 0.0 )
```

Comment:
I'm not sure to understand the meaning of this `if` here.

Reply:
Yes, here I just followed the structure of the existing poromechanics kernels. This is a small optimization for cases in which gravity is ignored. Honestly, this is not needed and we could remove the `if` everywhere in the poromechanics kernels (in another PR)

Reply:
But what if for any case we have `m_gravityAcceleration` negative because of some experimental setup of different axis convention? Is it properly considered?

Reply:
No, this case is not properly considered in the poromechanics kernels. Let's just remove this check from all the poromechanics kernels to avoid this type of confusion

Reply:
I bet the original check was `m_gravityAcceleration != 0.` but there was a warning w.r.t. equality comparisons for floating points. 🤣

Reply:
Done, I removed the `if( m_gravityAcceleration > 0.0 )` from the poromechanics kernels